drivers/dma/dma_mcux_lpc.c

```diff
@@ -312,13 +345,53 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 			data->descriptor_index = data->descriptor_used;
 		}
 	} else {
-		/* block_count shall be 1 */
-		/* Only one buffer, enable interrupt */
-		xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
-					config->dest_data_size,
-					src_inc,
-					dst_inc,
-					block_config->block_size);
+		/* If DMA Channel Chaining is expected, then the
+		 * burst length must be setup differently as well.
+		 */
+		if (!config->dest_chaining_en && !config->source_chaining_en) {
+			k_sem_give(&is_trig_mux_in_use);
+			/* block_count shall be 1, in other words, this
+			 * channel will transfer the full length of data.
+			 */
+			xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
+						      config->dest_data_size,
+						      src_inc,
+						      dst_inc,
+						      block_config->block_size);
+		}
+
+		else if(config->dest_chaining_en && config->source_chaining_en) {
+			/* block_count shall be 1, in other words, this
+			 * channel will transfer the full length of data.
+			 */
+			xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
+						      config->dest_data_size,
+						      src_inc,
+						      dst_inc,
+						      block_config->block_size);
+		} else if (config->dest_chaining_en) {
+			/* block_count shall be a portion of the transfer
+			 * and is assumed to be the length that is passed in
+			 * via config->source_burst_length
+			 */
+			xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
+						config->dest_data_size,
+						src_inc,
+						dst_inc,
+						config->source_burst_length);
+		} else if (config->source_chaining_en) {
+			/* In this case, a portion of the data has been filled
+			 * already thus the block_count is assumed to be the
+			 * full length of data minus what is passed in via
+			 * config->source_burst_length.
+			 */
+			xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
+						config->dest_data_size,
+						src_inc,
+						dst_inc,
+						block_config->block_size -
+						config->source_burst_length);
+		}
```

Comment:
This portion looks incorrect as we are not transferring the amount of data the user has provided via `block_size`. My suggestion would be to focus on channel chaining to trigger an event of transfer completion.

Reply:
this does look incorrect and should be addressed in the future of how to handle burst lengths